contracts/bprotocol/Import.sol

```diff
@@ -0,0 +1,227 @@
+
+        address cETH = registry.cEther();
+        bETH = bComptroller.c2b(cETH);
+    }
+
+    function repayAllDebt(address[] memory cTokenDebt,
+                          address[] memory debtUnderlying,
+                          uint[] memory originalDebt,
+                          address account) internal {
+        for(uint i = 0 ; i < cTokenDebt.length ; i++) {
+          address cDebt = cTokenDebt[i];
+          uint debt = originalDebt[i];
+          if(debtUnderlying[i] == ETH) {
+            ICEther(cDebt).repayBorrowBehalf.value(debt)(account);
+          }
+          else {
+            IERC20(debtUnderlying[i]).safeApprove(address(cDebt), debt);
+            require(0 == ICErc20(cDebt).repayBorrowBehalf(account, debt), "repay-failed");
+          }
+        }
+    }
+
+    function redeemAndDepositCollateral(address[] memory cTokenCollateral,
+                                        address[] memory collateralUnderlying,
+                                        address account,
+                                        address avatar) internal {
+
+        for(uint i = 0 ; i < cTokenCollateral.length ; i++) {
+          ICToken cColl = ICToken(cTokenCollateral[i]);
+          uint collBalance = cColl.balanceOf(account);
+          require(cColl.transferFrom(account, address(this), collBalance), "transferFrom-failed");
+          //IERC20(address(cColl)).safeTransferFrom(account, address(this), collBalance);
+          require(0 == cColl.redeem(collBalance), "redeem-failed");
+
+          address bColl = bComptroller.c2b(cTokenCollateral[i]);
+          if(collateralUnderlying[i] == ETH) {
+            BEther(bColl).mintOnAvatar.value(address(this).balance)(avatar);
+          }
+          else {
+            IERC20 token = IERC20(collateralUnderlying[i]);
+            uint tokenBalance = token.balanceOf(address(this));
+            token.safeApprove(bColl, tokenBalance);
+            require(0 == BErc20(bColl).mintOnAvatar(avatar, tokenBalance), "mint-failed");
+          }
+        }
+    }
+
+    function borrowDebtOnB(address[] memory cTokenDebt,
+                           uint[]    memory originalDebt,
+                           address avatar) internal {
+        for(uint i = 0 ; i < cTokenDebt.length ; i++) {
+          uint debt = originalDebt[i];
+          address bTokenDebt = bComptroller.c2b(cTokenDebt[i]);
+          require(0 == BErc20(bTokenDebt).borrowOnAvatar(avatar, debt), "borrowOnAvatar-failed");
+        }
+    }
+
+    // this is safe only if the owner of the compound account is an EOA
+    function importAccount(address[] calldata cTokenCollateral,
+                           address[] calldata collateralUnderlying,
+                           address[] calldata cTokenDebt,
+                           address[] calldata debtUnderlying,
+                           uint      ethFlashLoan) external {
+
+        require(cTokenCollateral.length == collateralUnderlying.length, "collateral-length-missmatch");
```

Comment:
its optional, but we can also check that the length of array is greater then 1 (or non zero)